src/tools/idlc/src/generator.c

```diff
@@ -357,8 +357,13 @@ idlc_generate(const idl_pstate_t *pstate)
 
   file = sep ? sep + 1 : path;
   if (idl_isabsolute(path) || !sep)
-    dir = empty;
-  else if (!(dir = idl_strndup(path, (size_t)(sep-path))))
+    cnt = idl_asprintf(&dir, "%.*s", idlc_config->prefix);
```

Comment:
Doesn't `%.*s` expect an integer? I don't think this is working when the `idlc` is invoked with absolute path

Reply:
Yep, that's exactly why an absolute path gives problems currently. I have fixed it locally, but I was more interested in whether this works for the ROS usecase. Expect a mergeable version soon. Thanks!